doc/index.rst

```diff
@@ -6,10 +6,45 @@
 Welcome to Sphinx-Gallery's documentation!
 ==========================================
 
-Sphinx extension for automatic generation of an example gallery.
+A Sphinx extension that builds an HTML version of any python
+script and puts it into an examples gallery.
+
 It is extracted from the scikit-learn project and aims to be an
 independent general purpose extension.
 
+Why do we need it?
+------------------
+
+* Because we have a lot of code that does something good enough for
+  someone to learn about it.
+* Looking to the visual output is simpler to our minds than reading
+  the source code.
+* The Gallery workflow:
+   * Search for something that more or less looks to what you want
+   * Learn about it
+
```

Comment:
I would suggest to add a section here called 'Features' with the following content:

* Runs Python files and captures outputs + figures
* Manages an arbitrary number of galleries
* Automatically links back from code in the gallery to the corresponding function documentation
* Automatically links from documentation of functions to examples using the function
* Notebook-style examples: turn your scripts into a full narrative

Where each of the items should have a link to a page of the documentation showcasing the item and explaining how to do it.